bootstrapper/internal/interfaces/interfaces.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package interfaces
+
+import (
+	"net"
+)
+
+// GetNetworkInterfaces retrieves all network interface addresses.
+func GetNetworkInterfaces() ([]string, error) {
+	var principalList []string
+	ifaces, err := net.Interfaces()
```

Comment:
If you pass `[]net.Interface` as input to the function, you should be able to add some unit tests.